src/tasks/checkEnvironment.js

```diff
@@ -31,12 +36,17 @@ var self = {
 
     // check for windows or Mac OSX
     if (['win32', 'darwin'].indexOf(process.platform) > -1) {
-      return self.boot2dockerExists();
+      self.dockerMachine = self.dockerMachineExists();
+      if (!self.dockerMachine) {
+        self.boot2docker = self.boot2dockerExists();
+      }
+      return self.dockerMachine || self.boot2docker;
```

Comment:
In that case, could I suggest:

```js
return self.dockerMachineExists() ? self.usingDockerMachine = true : self.boot2dockerExists()
```
I don't see that `self.boot2docker` is ever being used, so it at least seems unnecessary.